interactions/client/bot.py

```diff
@@ -493,6 +505,50 @@ async def __sync(self) -> None:  # sourcery no-metrics
             self.__guild_commands[_id] = {"commands": _cmds, "clean": True}
             __check_guild_commands[_id] = [cmd["name"] for cmd in _cmds] if _cmds else []
 
+        for cmd in self._commands:
+            if cmd.resolved:
+                continue
+
+            if cmd.default_scope and self._default_scope:
+                cmd.scope = (
+                    cmd.scope.extend(cmd.default_scope) if cmd.scope else self._default_scope
+                )
+
+            data: Union[dict, List[dict]] = cmd.full_data
+            coro = cmd.dispatcher
+
+            self.__check_command(
+                command=ApplicationCommand(**(data[0] if isinstance(data, list) else data)),
+                coro=coro,
+            )
+
+            if cmd.autocompletions:
+                self.__name_autocomplete.update(cmd.autocompletions)
+
+            coro = coro.__func__ if hasattr(coro, "__func__") else coro
+
+            coro._command_data = data
+            coro._name = cmd.name
+
+            if not (data["name"] if isinstance(data, dict) else data[0]["name"]) in (
+                (
+                    c._command_data["name"]
+                    if isinstance(c._command_data, dict)
+                    else c._command_data[0]["name"]
+                )
+                for c in self.__command_coroutines
+            ):
+                self.__command_coroutines.append(coro)
+
+            if cmd.scope not in (MISSING, None):
+                if isinstance(cmd.scope, List):
+                    [self._scopes.add(_ if isinstance(_, int) else _.id) for _ in cmd.scope]
+                else:
+                    self._scopes.add(cmd.scope if isinstance(cmd.scope, int) else cmd.scope.id)
+
+            self.event(coro, name=f"command_{cmd.name}")
+            cmd.resolved = True
+
```

Comment:
Imho this does not really belong into sync itself but into the _ready (if sync is off nothing would happen anyways and the command wouldn't be registered even if it is already synced)

Reply:
The thing is, Extensions call __sync after being loaded by default, so having it in __sync makes sure that commands that were added from Extensions are synced properly
I *could* make it so when Extensions are loaded it appends the commands and also modifies command coroutines, but that is just more work and less dynamic

Reply:
But if you have registered commands and disable sync the coros won't work because event is not called

Reply:
Ohh I haven't tested with disabled sync
Should I make a method named something like __resolve_commands and call it both in _ready and __sync?

Reply:
Only ready and extension load, sync should have nothing to do with it

Reply:
Ok I see
I will resolve this once I've done that, thanks a lot for the review

Reply:
Done